src/ui.ts

```diff
@@ -1652,6 +1626,39 @@ class ViewerSettings extends Panel {
     }
 }
 
+class XRSettings extends Panel {
+    private enableVRCheckbox: Checkbox;
+    private webXRContext: WebXRContext;
+
+    constructor(private ui: UI, private viewer: Viewer.Viewer) {
+        super();
+
+        this.webXRContext = viewer.webXRContext;
+
+        this.setTitle(VR_ICON, 'XR Settings');
+
+        this.contents.style.lineHeight = '36px';
+
+        this.enableVRCheckbox = new Checkbox('Enable XR?');
+        this.enableVRCheckbox.onchanged = () => { GlobalSaveManager.saveSetting(`EnableVR`, this.enableVRCheckbox.checked); };
+        this.contents.appendChild(this.enableVRCheckbox.elem);
+        GlobalSaveManager.addSettingListener('EnableVR', this.enableVRChecked.bind(this));
+    }
+
+    private enableVRChecked(saveManager: SaveManager, key: string): void {
+        const enableVR = saveManager.loadSetting<boolean>(key, false);
+        this.enableVRCheckbox.setChecked(enableVR);
+
+        if (this.enableVRCheckbox.checked) {
+            this.webXRContext.start();
+            this.viewer.setCameraController(XRCameraController);
```

Comment:
I'll probably split this out to allow the viewer to use one camera controller for the screen and one for all the XR stuff.